src/algebra/pointwise.lean

```diff
@@ -532,3 +535,54 @@ le_antisymm
   (by conv_rhs { rw [← closure_eq H, ← closure_eq K] }; apply closure_mul_le)
 
 end submonoid
+
+namespace group
+
+lemma card_pow_eq_card_pow_card_univ_aux {f : ℕ → ℕ} (h1 : monotone f)
+  {B : ℕ} (h2 : ∀ n, f n ≤ B) (h3 : ∀ n, f n = f (n + 1) → f (n + 1) = f (n + 2)) :
+  ∀ k, B ≤ k → f k = f B :=
+begin
+  have key : ∃ n : ℕ, n ≤ B ∧ f n = f (n + 1),
+  { contrapose! h2,
+    suffices : ∀ n : ℕ, n ≤ B + 1 → n ≤ f n,
+    { exact ⟨B + 1, this (B + 1) (le_refl (B + 1))⟩ },
+    exact λ n, nat.rec (λ h, nat.zero_le (f 0)) (λ n ih h, lt_of_le_of_lt (ih (n.le_succ.trans h))
+      (lt_of_le_of_ne (h1 n.le_succ) (h2 n (nat.succ_le_succ_iff.mp h)))) n },
+  { obtain ⟨n, hn1, hn2⟩ := key,
+    replace key : ∀ k : ℕ, f (n + k) = f (n + k + 1) ∧ f (n + k) = f n :=
+    λ k, nat.rec ⟨hn2, rfl⟩ (λ k ih, ⟨h3 _ ih.1, ih.1.symm.trans ih.2⟩) k,
+    replace key : ∀ k : ℕ, n ≤ k → f k = f n :=
+    λ k hk, (congr_arg f (nat.add_sub_cancel' hk)).symm.trans (key (k - n)).2,
+    exact λ k hk, (key k (hn1.trans hk)).trans (key B hn1).symm },
+end
+
+variables {G : Type*} [group G] [fintype G] (S : set G)
+
+open_locale classical
```

Comment:
Can you use `classical` inside the proof instead?

Reply:
I don't think so, because the `fintype` instances in the lemma statement won't work.

Reply:
Then that probably means you're missing a `decidable` assumption that ought to be in the lemma statement.

Reply:
I would guess `[decidable_pred (\mem S)]`

Reply:
`Π k : ℕ, decidable_pred (∈ (S ^ k))]` is enough

Reply:
It seems like `key` doesn't work with that assumption. Would something like `[Π s : set G, decidable_pred (∈ s)]` make sense?

Reply:
No, at that point there's no point in the assumption, as that's equivalent to classical.